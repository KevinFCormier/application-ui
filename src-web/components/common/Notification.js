/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2016, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use_strict'

import React from 'react'
import config from '../../../lib/shared/config'

class Notification extends React.Component {
  /* FIXME: Please fix disabled eslint rules when making changes to this file. */
  /* eslint-disable react/prop-types, react/jsx-no-bind */

  constructor(props) {
    super(props)
    this.state = {
      hide: false,
    }
  }

  render() {
    if (!this.state.hide) {
      const { type } = this.props
      return (
        <div className={`bx--inline-notification bx--inline-notification--${type}`}
          role="alert">
          <div className="bx--inline-notification__details">
            <svg className="bx--inline-notification__icon" aria-label="close">
              <use
                xlinkHref={`${
                  config.contextPath
                }/graphics/carbon-icons.svg#icon--${
                  type === 'success' ? 'checkmark' : type
                }--glyph`}
              />
            </svg>
            <div className="bx--inline-notification__text-wrapper">
              <p className="bx--inline-notification__title">
                {this.props.title}
              </p>
              {(() => {
                if (this.props.renderNotificationContent) {
                  return this.props.renderNotificationContent()
                } else {
                  return (
                    <p
                      className="bx--inline-notification__subtitle"
                      dangerouslySetInnerHTML={{
                        __html: this.props.description,
                      }}
                    />
                  )
                }
              })()}
            </div>
          </div>
          {(() => {
            if (this.props.allowClose) {
              return (
                <div>
                  <button
                    className="bx--inline-notification__close-button"
                    type="button"
                    onClick={this._handleClick.bind(this)}
                  >
                    <svg
                      className="bx--inline-notification__close-icon"
                      aria-label="close"
                    >
                      <use
                        xlinkHref={`${
                          config.contextPath
                        }/graphics/carbon-icons.svg#icon--close`}
                      />
                    </svg>
                  </button>
                </div>
              )
            }
          })()}
        </div>
      )
    }
    return <div />
  }

  _handleClick() {
    if (this.props.closed) {
      this.props.closed()
    } else {
      this.setState({
        hide: true,
      })
    }
  }
}
export default Notification
