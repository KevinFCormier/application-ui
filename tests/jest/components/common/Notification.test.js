/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
"use strict";

import React from "react";
import Notification from "../../../../src-web/components/common/Notification";
import renderer from "react-test-renderer";

describe("Notification component", () => {
  it("renders as expected", () => {
    const component = renderer.create(<Notification />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders as expected", () => {
    const props = {
      allowClose: true,
      renderNotificationContent() {}
    };
    const component = renderer.create(<Notification {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
