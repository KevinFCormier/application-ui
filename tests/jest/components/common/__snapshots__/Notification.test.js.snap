// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component renders as expected 1`] = `
<div
  className="bx--inline-notification bx--inline-notification--undefined"
  role="alert"
>
  <div
    className="bx--inline-notification__details"
  >
    <svg
      aria-label="close"
      className="bx--inline-notification__icon"
    >
      <use
        xlinkHref="undefined/graphics/carbon-icons.svg#icon--undefined--glyph"
      />
    </svg>
    <div
      className="bx--inline-notification__text-wrapper"
    >
      <p
        className="bx--inline-notification__title"
      />
      <p
        className="bx--inline-notification__subtitle"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Notification component renders as expected 2`] = `
<div
  className="bx--inline-notification bx--inline-notification--undefined"
  role="alert"
>
  <div
    className="bx--inline-notification__details"
  >
    <svg
      aria-label="close"
      className="bx--inline-notification__icon"
    >
      <use
        xlinkHref="undefined/graphics/carbon-icons.svg#icon--undefined--glyph"
      />
    </svg>
    <div
      className="bx--inline-notification__text-wrapper"
    >
      <p
        className="bx--inline-notification__title"
      />
    </div>
  </div>
  <div>
    <button
      className="bx--inline-notification__close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-label="close"
        className="bx--inline-notification__close-icon"
      >
        <use
          xlinkHref="undefined/graphics/carbon-icons.svg#icon--close"
        />
      </svg>
    </button>
  </div>
</div>
`;
